app/lib/common/assetGatewayMixin.js

```diff
@@ -0,0 +1,97 @@
+import React from "react";
+import Translate from "react-translate-component";
+import Icon from "../../components/Icon/Icon";
+
+function _getAvailableGateways(selectedAsset){
+    let {gatewayStatus} = this.state;
+
+    for (let g in gatewayStatus) { gatewayStatus[g].enabled = false; }
+
+    for (let g in gatewayStatus) {
+        this.props.backedCoins.get(g.toUpperCase(), []).find(c => {
+            if(g == "OPEN" && selectedAsset == c.backingCoinType && c.depositAllowed && c.isAvailable) { gatewayStatus.OPEN.enabled = true; }
+            if(g == "RUDEX" && selectedAsset == c.backingCoin && c.depositAllowed) { gatewayStatus.RUDEX.enabled = true; }
+        });
+    }
+    return gatewayStatus;
+}
+
+function _openGatewaySite() {
+    let {selectedGateway, gatewayStatus} = this.state;
+    let win = window.open(gatewayStatus[selectedGateway].support_url, "_blank");
+    win.focus();
+}
+
+function _getNumberAvailableGateways(){
```

Comment:
This needs to accommodate if the user has a withdrawable asset or not for the gateway as well.

Reply:
You're going to need to help me on that